Test/AwsCommonRuntimeKitTests/crt/FutureTests.swift

```diff
@@ -0,0 +1,27 @@
+//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//  SPDX-License-Identifier: Apache-2.0.
```

Comment:
review this file

Reply:
Should probably add some tests in there where the future waits in the test thread and is fulfilled from another thread. Should also have a test where multiple threads are waiting on the future.

Reply:
tests added